app/decorators/odk/condition_group_decorator.rb

```diff
@@ -0,0 +1,21 @@
+module Odk
+  class ConditionGroupDecorator < BaseDecorator
+
+    def to_odk
+      conjuction = (true_if == "all_met") ? I18n.t("common.and") : I18n.t("common.or")
+      result = decorated_members.map{ |m| "(#{m.to_odk})"}.join(" #{conjuction} ")
+      if negate?
+        result = "not(#{result})"
+      end
+      decorated_members.count == 0 ? nil : result
```

Comment:
This would typically go as a guard clause at the top instead of at the bottom so you save the effort of all the computation.